FlakyTest: tests/chainer_tests/test_gradient_check.py

`TestCheckBackwardNoneConvention_param_0_{dtype=None, size=3}.test_multiple_output`  is flaky
[Jenkins](https://jenkins.preferred.jp/job/chainer/job/chainer_pr/1872/TEST=chainer-py35,label=mn1-p100/console)
```
20:50:31 self = <chainer.testing._bundle.TestCheckBackwardNoneConvention_param_0_{dtype=None, size=3} testMethod=test_multiple_output>
20:50:31 
20:50:31     def test_multiple_output(self):
20:50:31         size = self.size
20:50:31         x1 = numpy.arange(size).astype('f')
20:50:31         x2 = numpy.arange(size).astype('f')
20:50:31         g1 = numpy.ones(size, dtype='f')
20:50:31         g2 = numpy.ones(size, dtype='f')
20:50:31     
20:50:31         def f(x, y):
20:50:31             s, t = IdentNoneIsZero()(x, y)
20:50:31             return s, t
20:50:31     
20:50:31         gradient_check.check_backward(
20:50:31 >           f, (x1, x2), (g1, g2), dtype=self.dtype, atol=1e-4, rtol=1e-3)
20:50:31 
20:50:31 f          = <function TestCheckBackwardNoneConvention.test_multiple_output.<locals>.f at 0x7efd83f7e378>
20:50:31 g1         = array([ 1.,  1.,  1.], dtype=float32)
20:50:31 g2         = array([ 1.,  1.,  1.], dtype=float32)
20:50:31 self       = <chainer.testing._bundle.TestCheckBackwardNoneConvention_param_0_{dtype=None, size=3} testMethod=test_multiple_output>
20:50:31 size       = 3
20:50:31 x1         = array([ 0.,  1.,  2.], dtype=float32)
20:50:31 x2         = array([ 0.,  1.,  2.], dtype=float32)
20:50:31 
20:50:31 tests/chainer_tests/test_gradient_check.py:690: 
20:50:31 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
20:50:31 chainer/gradient_check.py:900: in check_backward
20:50:31     detect_nondifferentiable, is_immutable_params=False
20:50:31 chainer/gradient_check.py:464: in run
20:50:31     self._run()
20:50:31 chainer/gradient_check.py:507: in _run
20:50:31     self._compare_gradients(gx_numeric, gx_backward, directions)
20:50:31 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
20:50:31 
20:50:31 self = <chainer.gradient_check._CheckBackward object at 0x7efd8c0cd550>
20:50:31 gx_numeric = array(0.01885576784843579), gx_backward = 0.018735989313901857
20:50:31 directions = [array([-0.54368434,  0.25842544, -0.55366077]), array([ 0.08170945,  0.27989749,  0.49604873])]
20:50:31 
20:50:31     def _compare_gradients(self, gx_numeric, gx_backward, directions):
20:50:31         atol = self.atol
20:50:31         rtol = self.rtol
20:50:31         # Compare the gradients
20:50:31         try:
20:50:31             testing.assert_allclose(
20:50:31                 gx_numeric, gx_backward, atol=atol, rtol=rtol)
20:50:31         except AssertionError as e:
20:50:31             eps = self.eps
20:50:31             xs = self.xs
20:50:31             gys = self.gys
20:50:31             f = six.StringIO()
20:50:31             f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
20:50:31                 eps, atol, rtol))
20:50:31             for i, x in enumerate(xs):
20:50:31                 f.write('inputs[{}]:\n'.format(i))
20:50:31                 f.write('{}\n'.format(x))
20:50:31             for i, gy in enumerate(gys):
20:50:31                 f.write('grad_outputs[{}]:\n'.format(i))
20:50:31                 f.write('{}\n'.format(gy))
20:50:31             for i, d in enumerate(directions):
20:50:31                 f.write('directions[{}]:\n'.format(i))
20:50:31                 f.write('{}\n'.format(d))
20:50:31             f.write('gradients (numeric):  {}\n'.format(gx_numeric))
20:50:31             f.write('gradients (backward): {}\n'.format(gx_backward))
20:50:31             f.write('\n')
20:50:31             f.write('x: numeric gradient, y: backward gradient')
20:50:31             f.write(str(e))
20:50:31 >           raise AssertionError(f.getvalue())
20:50:31 E           AssertionError: Parameterized test failed.
20:50:31 E           
20:50:31 E           Base test method: TestCheckBackwardNoneConvention.test_multiple_output
20:50:31 E           Test parameters:
20:50:31 E             dtype: None
20:50:31 E             size: 3
20:50:31 E           
20:50:31 E           
20:50:31 E           (caused by)
20:50:31 E           AssertionError: check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
20:50:31 E           inputs[0]:
20:50:31 E           [ 0.  1.  2.]
20:50:31 E           inputs[1]:
20:50:31 E           [ 0.  1.  2.]
20:50:31 E           grad_outputs[0]:
20:50:31 E           [ 1.  1.  1.]
20:50:31 E           grad_outputs[1]:
20:50:31 E           [ 1.  1.  1.]
20:50:31 E           directions[0]:
20:50:31 E           [-0.54368434  0.25842544 -0.55366077]
20:50:31 E           directions[1]:
20:50:31 E           [ 0.08170945  0.27989749  0.49604873]
20:50:31 E           gradients (numeric):  0.01885576784843579
20:50:31 E           gradients (backward): 0.018735989313901857
20:50:31 E           
20:50:31 E           x: numeric gradient, y: backward gradient
20:50:31 E           Not equal to tolerance rtol=0.001, atol=0.0001
20:50:31 E           
20:50:31 E           (mismatch 100.0%)
20:50:31 E            x: array(0.01885576784843579)
20:50:31 E            y: array(0.018735989313901857)
20:50:31 E           
20:50:31 E           assert_allclose failed: 
20:50:31 E             shape: () ()
20:50:31 E             dtype: float64 float64
20:50:31 E             i: (0,)
20:50:31 E             x[i]: 0.01885576784843579
20:50:31 E             y[i]: 0.018735989313901857
20:50:31 E             relative error[i]: 0.0063929655662782565
20:50:31 E             absolute error[i]: 0.00011977853453393195
20:50:31 E             relative tolerance * |y[i]|: 1.8735989313901857e-05
20:50:31 E             absolute tolerance: 0.0001
20:50:31 E             total tolerance: 0.00011873598931390186
20:50:31 E           x: 0.01885576784843579
20:50:31 E           y: 0.018735989313901857
20:50:31 
20:50:31 atol       = 0.0001
20:50:31 d          = array([ 0.08170945,  0.27989749,  0.49604873])
20:50:31 directions = [array([-0.54368434,  0.25842544, -0.55366077]), array([ 0.08170945,  0.27989749,  0.49604873])]
20:50:31 eps        = 0.001
20:50:31 f          = <_io.StringIO object at 0x7efd8c0a81f8>
20:50:31 gx_backward = 0.018735989313901857
20:50:31 gx_numeric = array(0.01885576784843579)
20:50:31 gy         = array([ 1.,  1.,  1.], dtype=float32)
20:50:31 gys        = (array([ 1.,  1.,  1.], dtype=float32), array([ 1.,  1.,  1.], dtype=float32))
20:50:31 i          = 1
20:50:31 rtol       = 0.001
20:50:31 self       = <chainer.gradient_check._CheckBackward object at 0x7efd8c0cd550>
20:50:31 x          = array([ 0.,  1.,  2.], dtype=float32)
20:50:31 xs         = (array([ 0.,  1.,  2.], dtype=float32), array([ 0.,  1.,  2.], dtype=float32))
20:50:31 
20:50:31 chainer/gradient_check.py:537: AssertionError
```

--- a/tests/chainer_tests/test_gradient_check.py
+++ b/tests/chainer_tests/test_gradient_check.py
@@ -677,21 +677,21 @@ class TestCheckBackwardNoneConvention(unittest.TestCase):
 
     def test_multiple_output(self):
         size = self.size
-        x1 = numpy.arange(size).astype('f')
-        x2 = numpy.arange(size).astype('f')
-        g1 = numpy.ones(size, dtype='f')
-        g2 = numpy.ones(size, dtype='f')
+        x1 = numpy.arange(size).astype('float32')
+        x2 = numpy.arange(size).astype('float32')
+        g1 = numpy.ones(size, dtype='float32')
+        g2 = numpy.ones(size, dtype='float32')
 
         def f(x, y):
             s, t = IdentNoneIsZero()(x, y)
             return s, t
 
         gradient_check.check_backward(
-            f, (x1, x2), (g1, g2), dtype=self.dtype, atol=1e-4, rtol=1e-3)
+            f, (x1, x2), (g1, g2), dtype=self.dtype, atol=1e-2, rtol=1e-2)
         gradient_check.check_backward(
-            f, (x1, x2), (g1, None), dtype=self.dtype, atol=1e-4, rtol=1e-3)
+            f, (x1, x2), (g1, None), dtype=self.dtype, atol=1e-2, rtol=1e-2)
         gradient_check.check_backward(
-            f, (x1, x2), (None, g2), dtype=self.dtype, atol=1e-4, rtol=1e-3)
+            f, (x1, x2), (None, g2), dtype=self.dtype, atol=1e-2, rtol=1e-2)
 
 
 class TestCheckBackwardFailure(unittest.TestCase):

